front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -44,6 +44,18 @@ class PipelineController {
     pipelineDAO.getPipelinesByApplication(application)
   }
 
+  @RequestMapping(value = 'getById/{id:.+}', method = RequestMethod.GET)
```

Comment:
I'm not really following what is proposed.

I wouldn't want to see `/pipelines/id/{id}` or `/pipelines/application/{application}` or `/pipelines?id=xxx`.

Given that all these are additional, there's no real harm in:
```
/v2/pipelines/{id}
/v2/pipelines?name=xxx
```

You only really want to add the ability to fetch a pipeline by id or name ... and this seems the simplest thing to do. You would need to track down callers that should be using the fetch by id anyways.